nixos/modules/tasks/network-interfaces.nix

```diff
@@ -903,6 +903,11 @@ in
         Whether to use DHCP to obtain an IP address and other
         configuration for all network interfaces that are not manually
         configured.
+
+        Using this option is highly discouraged and also incompatible with
```

Comment:
Might be a bit far fetched but how do we plan to proceed with installer images if we discourage usage of this? Wouldn't it be best practice to ship the installer with an encouraged configuration? We should be a good example to users and not just tell them to not use it.

Reply:
Currently working on this

Reply:
Does `networking.interfaces."en*".useDHCP = true` work? reason I ask is because setting a wildcard for `matchConfig` is supported for networkd. 

I guess then for the installer image we can then set `networking.interfaces."en*".useDHCP = true` to get an IP address on all wired connections (seems like a sane default), but it won't try to ask for DHCP leases on bridges, veths etc

Reply:
IMHO we shouldn't solve this in this PR.

Reply:
@andir I understood your comment as changing nixos-generate-config, from the conversation I just had he misunderstood this, so I've added a commit to change nixos-generate-config. It adds `networking.useDHCP = false;` and adds `networking.interfaces.<name>.useDHCP = true;` for all interfaces is `/sys/class/net` except for `lo`. Virtual interfaces should not be present in our install media anyway.